mumax3c/scripts/util.py

```diff
@@ -81,22 +81,49 @@ def set_parameter(parameter, name, system):
 
     # Spatially varying parameter defined using subregions.
     elif isinstance(parameter, dict):
-        names = system.m.mesh.subregions.keys()
-        subregions_dict = dict(zip(names, range(len(names))))
-
         for key, value in parameter.items():
-            if isinstance(value, numbers.Real):
-                mx3 += f"{name}.setregion({subregions_dict[key]}, {value})\n"
-
-            elif isinstance(value, (list, tuple, np.ndarray)):
-                mx3 += (
-                    f"{name}.setregion({subregions_dict[key]}, "
-                    "vector({}, {}, {}))\n".format(*value)
-                )
+            if ":" in key:
+                mx3 += _set_inter_reg_params(key, value, name, system)
+            else:
+                for region in system.region_relator[key]:
+                    if isinstance(value, numbers.Real):
+                        mx3 += f"{name}.setregion({region}, {value})\n"
+                    elif isinstance(value, (list, tuple, np.ndarray)):
+                        mx3 += (
+                            f"{name}.setregion({region}, "
+                            "vector({}, {}, {}))\n".format(*value)
+                        )
 
     else:
         # In mumax3, the parameter cannot be set using Field.
         msg = f"Cannot use {type(parameter)} to set parameter."
         raise TypeError(msg)
 
     return mx3
+
+
+def _set_inter_reg_params(key, value, name, system):
+    sub_regions = key.split(":")
+    if name not in {"Aex", "Dind"}:  # ext_InterDbulk not available.
+        raise ValueError(
+            "Only Aex and Dind can be set for different region in mumax3."
+            f" Cannot set inter region {name}"
+        )
+    elif np.any([len(system.region_relator[sub_reg]) > 1 for sub_reg in sub_regions]):
+        raise ValueError(
```

Comment:
I think it would be good to implement this before release.

Reply:
Absolutely. I was thinking, the present work around should be good till the workshop and in the mean time we can think of a clever way to set inter region interaction befor release. What do you think?

Reply:
Didn't we have the impression that setting interactions between unrelated regions (with a space or diagonal) had essentially no effect? Maybe we could then just set the interaction between all pairs of regions.

Reply:
Yes, interactions between non-contiguous regions has no effect so we should be able to do all pairs of regions.

Reply:
@samjrholt and @lang-m thats promissing! I'll try to fix this then.

Reply:
I have addressed this in commit 6f160498f55527f8bec39aa4bc482c36f83306e6.

Reply:
Looks good to me

Reply:
@swapneelap nice implementation, I'm happy with it.